### PR TITLE
upgrade to centos 7 box and using ejabberd binary installer

### PR DESCRIPTION
**Pull request by PuppetNinja**

### CODE REVIEW

**Comment by CyrilPeponnet** [comment]
This looks awesome but few things. 

You don't need expect as you can give the binary installer all the args with the flag unattended. 

Please use the ejabberd from the wiki page.

File: manifests/central_server.pp [hunk 1]
```diff
@@ -3,64 +3,72 @@
...
   }
-
-  class { 'ejabberd':
-    config_content  => template('archipel/ejabberd.cfg.erb'),
-    package_ensure  => 'installed',
-    package_name    => 'ejabberd',
-    service_reload  => true,
+  include archipel
+  package { "expect":
+    ensure      => installed
```
Comment (CyrilPeponnet) [inline manifests/central_server.pp]: This is not required see below.

File: manifests/central_server.pp [hunk 2]
```diff
@@ -3,64 +3,72 @@
...
+    mode        => 0755
   }
   ->
-  file { "${ejabberd::params::lib_dir}/ebin/ejabberd_xmlrpc.beam":
-    ensure  => present,
-    source  => "/usr/local/src/ejabberd-modules/ejabberd_xmlrpc/trunk/ebin/ejabberd_xmlrpc.beam",
+  exec { "running the binary installer via expect script ":
+    command     => "expect /usr/local/bin/ejabberd_binary_install.exp",
+    user        => root,
+    timeout     => 0  #disable the timeout for binary installer
```
Comment (CyrilPeponnet) [inline manifests/central_server.pp]: Create a user to run ejabberd as non privilegied user (home located to /opt/ejabberd).
Install the binary as the ejabberd user ejabberd-15.0X-linux-x86_64-installer.run --mode unattended --adminpw admin --prefix /opt/ejabberd/

File: manifests/central_server.pp [hunk 3]
```diff
@@ -3,64 +3,72 @@
...
-  # we cannot use the puppett-ejabberd module functionality.
-  #ejabberd::contrib::module{ 'mod_xmlrpc':  }
-  include archipel
-  ejabberd_user { 'admin':
-    host        => 'central-server.archipel.priv',
-    password    => 'admin'
+  ->
+  package { ['erlang', 'erlang-xmlrpc']:
+    ensure      => present
+  }
```
Comment (CyrilPeponnet) [inline manifests/central_server.pp]: Not needed

File: manifests/init.pp [hunk 4]
```diff
@@ -16,19 +16,23 @@
...
-               "python-argparse", "python-pip"]:
+               "python-pip"]:
     }
-    service { 'iptables':
-      ensure => 'stopped',
-      enable => false,
+    ->
+    exec {"pip install argparse":
+      unless => "ls /usr/lib/python2.7/site-packages/argparse-*"
     }
```
Comment (CyrilPeponnet) [inline manifests/init.pp]: Not needed argpase is part of builtin libraries since 2.7

File: templates/ejabberd.yml.erb [hunk 5]
```diff
@@ -0,0 +1,663 @@
+###
+###               ejabberd configuration file
+###
+###
```
Comment (CyrilPeponnet) [inline templates/ejabberd.yml.erb]: Use the one from the wiki please :)

**Comment by PuppetNinja** [comment]
I got crash when running binary installer with ejabberd user in unattended mode, I am not sure why I should installing the ejabberd as user 'ejabberd', running with root privilege makes more sense to me actually, since it follows the yum and apt-get

Following is what I saw as user 'ejabberd'
bash-4.2$ ./ejabberdctl help
(no error logger present) error: "Error in process <0.2.0> with exit value: {undef,[{gen_event,start_link,[{local,error_logger}],[]},{error_logger,start_link,0,[{file,"
error_logger.erl"},{line,72}]},{init,start_in_kernel,5,[]},{init,eval_script,8,[]},{init,do_boot,3,[]}]}
"
{"init terminating in do_boot",{undef,[{gen_event,start_link,[{local,error_logger}],[]},{error_logger,start_link,0,[{file,"error_logger.erl"},{line,72}]},{init,start_in
_kernel,5,[]},{init,eval_script,8,[]},{init,do_boot,3,[]}]}}

Crash dump was written to: /opt/ejabberd/logs/erl_crash_20150825-001743.dump

**Comment by PuppetNinja** [comment]
And I run with root, it works actually pretty well...

**Comment by CyrilPeponnet** [comment]
This should work as expected...

Be sure to create the user and set it's homedir to /opt/ejabberd

```bash
groupadd -r ejabberd
useradd -r -m \
       -g ejabberd \
       -d /opt/ejabberd \
       -s /usr/sbin/nologin \
       ejabberd
```
then as ejabberd user

```bash
/tmp/ejabberd-installer.run \
            --mode unattended \
            --prefix /opt/ejabberd \
            --adminpw admin \
```

If it's not working, please open a ticket here: https://github.com/processone/ejabberd

**Comment by CyrilPeponnet** [comment]
maybe we can use the rpm as there is one now :)

https://www.process-one.net/en/ejabberd/downloads

**Comment by CyrilPeponnet** [comment]
@PuppetNinja just to let you know that I'm working on your PR to make it works :)

**Comment by PuppetNinja** [comment]
Merci :-)

**Comment by CyrilPeponnet** [comment]
Closing this one as I've done a separate PR, but Merci Chao for this work :)

**Pull request closed without merging**